File: UI/fitbook_desktop/lib/models/responses/user_membership_status_audit_response.dart
import 'package:json_annotation/json_annotation.dart';

import '../enums/membership_status.dart';

part 'user_membership_status_audit_response.g.dart';

@JsonSerializable()
class UserMembershipStatusAuditResponse {
  UserMembershipStatusAuditResponse({
    required this.id,
    required this.previousStatus,
    required this.newStatus,
    required this.changedAtUtc,
    this.reason,
    required this.changedByUserFullName,
  });

  final int id;
  final MembershipStatus previousStatus;
  final MembershipStatus newStatus;
  final DateTime changedAtUtc;
  final String? reason;
  final String changedByUserFullName;

  factory UserMembershipStatusAuditResponse.fromJson(Map<String, dynamic> json) =>
      _$UserMembershipStatusAuditResponseFromJson(json);

  Map<String, dynamic> toJson() => _$UserMembershipStatusAuditResponseToJson(this);
}

File: FitBook.Model/Responses/UserMemberships/UserMembershipStatusAuditResponse.cs
using FitBook.Model.Enums;

namespace FitBook.Model.Responses.UserMemberships;

public class UserMembershipStatusAuditResponse
{
    public int Id { get; set; }
    public MembershipStatus PreviousStatus { get; set; }
    public MembershipStatus NewStatus { get; set; }
    public DateTime ChangedAtUtc { get; set; }
    public string? Reason { get; set; }
    public string ChangedByUserFullName { get; set; } = string.Empty;
}
